### ISSUE TITLE
(Why not) use message batching?

### ISSUE THREAD

**Issue by SimonHeybrock** [scipp/beamlime#273]
One of the things Beamlime does is to consume, say, 14 `ev44` messages per second, push them through the internal message router, collect them for a time interval, concat and hand them to the data reduction handler.

Can we just use message batching? That might significantly simplify the logic.

**Comment by SimonHeybrock** [comment]
Not sure exactly how it would work in practice, but something like
```python
messages = detector_consumer.consume(num_messages=20, timeout=1.0)
```

**Comment by YooSunYoung** [comment]
If it's about consuming multiple messages at once, yes, we should probably do that.

But I didn't do that since that `timeout` blocks the whole process.

AOI kafka seems to have a better interface for that and it probably doesn't block the event loop:

https://aiokafka.readthedocs.io/en/stable/api.html#aiokafka.AIOKafkaConsumer

It also supports batch processing of messages but I seems like it only works if producer publishes messages in batch according to this documentation: https://docs.confluent.io/kafka/design/efficient-design.html ...?

**Comment by SimonHeybrock** [comment]
That is not how understood the mechanism. The consumer can decide how to fetch. Obviously, if we fetch too frequently we may only get a single message. But if we are willing to accept some more latency (which we also have with the "manual" beamlime-side batching) then we can fetch batches.

**Comment by YooSunYoung** [comment]
> The consumer can decide how to fetch.

Yes. What I said is, the `batch` messages of `kafka` and consuming multiple messages are two different concepts.

> But if we are willing to accept some more latency (which we also have with the "manual" beamlime-side batching) then we can fetch batches.

Yeah, but beamlime-side batching is to concatenate each fields of the dataset. How can it be handled by consumer itself...?

**Comment by SimonHeybrock** [comment]
Outdated.
